core/trino-main/src/test/java/io/trino/operator/aggregation/BenchmarkDecimalAggregation.java

```diff
@@ -82,6 +83,27 @@ public Block benchmarkEvaluateIntermediate(BenchmarkData data)
         return builder.build();
     }
 
+    @Benchmark
+    @OutputTimeUnit(TimeUnit.MICROSECONDS)
+    public Block benchmarkEvaluateIntermediateOnly(BenchmarkData data)
+    {
+        GroupedAggregator aggregator = data.getPartialAggregator();
+        BlockBuilder builder = aggregator.getType().createBlockBuilder(null, data.getGroupCount());
+        for (int groupId = 0; groupId < data.getGroupCount(); groupId++) {
+            aggregator.evaluate(groupId, builder);
+        }
+        return builder.build();
+    }
+
+    @Benchmark
+    @OutputTimeUnit(TimeUnit.MICROSECONDS)
+    public Object benchmarkDeserialize(BenchmarkData data)
```

Comment:
Why it's called `benchmarkDeserialize`? I think `processPage` does more than just deserialization

Reply:
Still, for a code reader this is confusing.

> there is no way to benchmark deserialize only so

Sure there is: `LongDecimalWithOverflowAndLongStateSerializer` is a class and there is nothing stopping us from benchmarking it directly